dev/list_changed_files.py

```diff
@@ -0,0 +1,44 @@
+"""
+A python script to list changed files in a specified pull request.
+
+Usage:
+---------------------------------------------------------------------------
+# List changed files in https://github.com/mlflow/mlflow/pull/3191
+$ python dev/list_changed_files.py --repository mlflow/mlflow --pr-num 3191
+---------------------------------------------------------------------------
+"""
+import argparse
+import os
+import requests
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--repository", help="Owner and repository name", required=True)
+    parser.add_argument("--pr-num", help="Pull request number", required=True)
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+    changed_files = []
+    per_page = 100
+    page = 1
+    token = os.environ.get("GITHUB_TOKEN")
+    headers = {"Authorization": f"token {token}"} if token else None
+    # Ref: https://docs.github.com/en/rest/reference/pulls#list-pull-requests-files
+    url = f"https://api.github.com/repos/{args.repository}/pulls/{args.pr_num}/files"
+    while True:
+        resp = requests.get(url, params={"per_page": per_page, "page": page}, headers=headers)
```

Comment:
If there are 100 files changed, then in the second round of the loop, `page=2`. Would the `resp` be empty or would it raise an error? If `resp` is empty, I guess `files` would be 0 and then execute`break` and there is no problem handling this case.

Reply:
In that case, `resp.json()` returns an empty list:

https://api.github.com/repos/mlflow/mlflow/pulls/5041/files?per_page=100&page=2